mlrun/api/db/sqldb/db.py

```diff
@@ -3916,3 +3918,104 @@ def set_run_notifications(
                 project=project,
             )
         self._commit(session, [run], ignore=True)
+
+    def store_datastore_profile(
+        self, session, info: mlrun.common.schemas.DatastoreProfile
+    ):
+        """
+        Create or replace a datastore profile.
+        :param session: SQLAlchemy session
+        :param info: datastore profile
+        :returns: None
+        """
+        info.project = info.project or config.default_project
+        profile = self._query(
+            session, DatastoreProfile, name=info.name, project=info.project
+        )
+        first = profile.first()
```

Comment:
Better to use -
`profile = self._query(...).one_or_none()`
Which will always return either `None` or the actual profile object if exists. It will also assert if there are several records returned from the query, which is a good sanity in this case.